x/superfluid/keeper/stake_test.go

```diff
@@ -1028,6 +1031,548 @@ func (s *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUnbondConvertAndStake() {
+	defaultJoinTime := s.Ctx.BlockTime()
+	type tc struct {
+		notSuperfluidDelegated bool
+		superfluidUndelegating bool
+		unlocking              bool
+		unlocked               bool
+		expectedError          error
+	}
+	testCases := map[string]tc{
+		"lock that is superfluid delegated": {},
+		"lock that is superfluid undelegating": {
+			superfluidUndelegating: true,
+		},
+		"bonded lock, not superfluid delegated": {
+			notSuperfluidDelegated: true,
+		},
+		"lock that is unlocking": {
+			unlocking:              true,
+			superfluidUndelegating: true,
+		},
+		"unlocked gamm shares": {
+			notSuperfluidDelegated: true,
+			unlocked:               true,
+		},
```

Comment:
Do we have a test with CL locks somewhere?

Reply:
Those cases are tested respectively each in TestConvertLockToStake and TestConvertUnlockedToStake

Reply:
I see this:
https://github.com/osmosis-labs/osmosis/blob/ba1d94388f923880c7a956407855d7c66caf2f4c/x/superfluid/keeper/stake_test.go#L1142-L1148

But I do not see a CL lock explicitly tested.

It would be valuable to set up a separate test case where CL lock is created and we show that converting it fails.